src/locked_dataset.hpp

```diff
@@ -681,13 +681,24 @@ class locked_dataset
             char const *options_array[1 << 8];
             GDALWarpAppOptions *app_options = nullptr;
 
+            bool output_format_configured = false;
             unsigned int i = 0;
             for (i = 0; i < options_vector.size(); ++i)
             {
-                options_array[i] = options_vector[i].c_str();
+                auto value = options_vector[i].c_str();
+                if(i % 2 == 0 && value == std::string("-of")) {
+                    output_format_configured = true;
+                }
+                options_array[i] = value;
             }
-            options_array[i++] = "-of";
-            options_array[i++] = "VRT";
+
+            // GDAL 3.9.x+ fails on duplicate arguments that can't be repeatable
+            // https://github.com/OSGeo/gdal/blob/v3.9.0/apps/argparse/argparse.hpp#L928-L931
+            if(!output_format_configured) {
+                options_array[i++] = "-of";
+                options_array[i++] = "VRT";
+            }
+
```

Comment:
I assume that this also work with version of GDAL earlier than 3.9.x?

Reply:
Yes! < 3.9 GDAL was ignorant to duplicate args, and was using the last one passed. Now they have a more clear semantics of what args can be repeatable

Reply:
:tada: